### ISSUE TITLE
你好 LineaRE/utils.py 代码 246行 有点疑惑

### ISSUE THREAD

**Issue by ljhry** [pengyanhui/LineaRE#6]
你好，配置文件中有alpha超参数，但是在实现loss函数时，这块注释掉了，这块有点小疑惑，这种实现方式是采用self adversarial negative sampling 策略吗
`  

    # neg_score = (func.softmax(-neg_score * config.alpha, dim=-1).detach()
    #              * func.softplus(-neg_score, beta=config.beta)).sum(dim=-1)
    neg_score = func.softplus(-neg_score, beta=config.beta)
    adv = neg_score.detach()
    neg_score = torch.sum(adv * neg_score, dim=-1) / torch.sum(adv, dim=-1)
`

**Comment by pengyanhui** [comment]
注释掉的部分是 self adversarial negative sampling 原文的，下面的是我尝试的小trick，结果证明没有什么差别，你用注释掉的部分吧，把下面的部分注释掉。或者你去new code文件夹里看新代码，是经过精心整理重构的，比较老代码清晰很多。

**Comment by ljhry** [comment]
非常感谢！还有一个疑问请教一下您，在new-code/lineare.py 62 和64 行
`score = wh * self.ent_embd.weight + (r - wt * t)`
这块测试的时候是用所有实体 self.ent_embd.weight，不应该是h吗，期待你的回复

**Comment by pengyanhui** [comment]
> 非常感谢！还有一个疑问请教一下您，在new-code/lineare.py 62 和64 行 `score = wh * self.ent_embd.weight + (r - wt * t)` 这块测试的时候是用所有实体 self.ent_embd.weight，不应该是h吗，期待你的回复

测试的时候是要把所有实体的 score 都计算的，计算好之后再进行排序。论文 IV.B 节有详细的说明。
